CMakeLists.txt

```diff
@@ -1,281 +1,294 @@
+check_type_size("long" SIZEOF_LONG)
+check_type_size("short" SIZEOF_SHORT)
 
-TEST_BIG_ENDIAN(HOST_BIG_ENDIAN)
+test_big_endian(HOST_BIG_ENDIAN)
 
-check_c_compiler_flag(-fvisibility=hidden EVHTP_HAS_VISIBILITY_HIDDEN)
-check_c_compiler_flag(-std=c99 EVHTP_HAS_C99)
-check_c_compiler_flag(-fsanitize=address EVHTP_HAS_SANITIZE_ADDRESS)
-check_c_compiler_flag(-fsanitize=leak EVHTP_HAS_SANITIZE_LEAK)
+check_c_compiler_flag(-std=c99 has_c99)
 
 # NOTE in as of ubuntu 14, this is enabled by default, so if you see it fail
 # the check, don't freak out.
-check_c_compiler_flag(-fstack-protector-strong EVHTP_HAS_STACK_PROTECTOR)
+check_c_compiler_flag(-fstack-protector-strong has_stack_protector)
 
-if (EVHTP_HAS_SANITIZE_ADDRESS)
-    set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fsanitize=address")
-endif()
+set(LIBEVHTP_SOURCE_FILES
+    evhtp.c
+    numtoa.c
+    parser.c)
 
-if (EVHTP_HAS_SANITIZE_LEAK)
-    set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fsanitize=leak")
+hunter_add_package(Libevent)
+find_package(Libevent CONFIG REQUIRED)
+list(APPEND LIBEVHTP_EXTERNAL_LIBS Libevent::event_core Libevent::event_extra)
+list(APPEND package_deps Libevent)
+set(public_headers
+  "include/evhtp.h"
+  "include/evhtp/evhtp.h"
+  "include/evhtp/parser.h")
+
+if(NOT EVHTP_DISABLE_SSL)
+    hunter_add_package(OpenSSL)
+    find_package(OpenSSL REQUIRED)
+    list(APPEND LIBEVHTP_SOURCE_FILES sslutils.c)
+    list(APPEND LIBEVHTP_EXTERNAL_LIBS OpenSSL::SSL OpenSSL::Crypto)
+    list(APPEND package_deps OpenSSL)
+    list(APPEND public_headers "include/evhtp/sslutils.h")
 endif()
 
-if (EVHTP_HAS_STACK_PROTECTOR)
-    set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fstack-protector-strong")
+if(NOT EVHTP_DISABLE_EVTHR)
+    find_package(Threads REQUIRED)
+    if(CMAKE_THREAD_LIBS_INIT)
+        list(APPEND LIBEVHTP_SOURCE_FILES  thread.c)
+        list(APPEND LIBEVHTP_EXTERNAL_LIBS Threads::Threads)
+        list(APPEND package_deps Threads)
+        list(APPEND public_headers "include/evhtp/thread.h")
+    endif()
 endif()
 
-if (EVHTP_HAS_C99)
-    add_definitions(-DEVHTP_HAS_C99)
+# TODO: Add Oniguruma to Hunter
+# if(NOT EVHTP_DISABLE_REGEX)
+#     find_package(Oniguruma CONFIG REQUIRED)
+#     list(APPEND LIBEVHTP_EXTERNAL_LIBS ${ONIGURUMA_LIBRARIES})
+#     list(APPEND package_deps Oniguruma)
+# endif()
+
+# TODO: Add jemalloc and tcmalloc to Hunter
+# if(EVHTP_ALLOCATOR STREQUAL "jemalloc")
+#     find_package(jemalloc CONFIG REQUIRED)
+#     list(APPEND LIBEVHTP_EXTERNAL_LIBS jemalloc)
+#     list(APPEND package_deps jemalloc)
+# elseif(EVHTP_ALLOCATOR STREQUAL "tcmalloc")
+#     find_package(tcmalloc CONFIG REQUIRED)
+#     list(APPEND LIBEVHTP_EXTERNAL_LIBS tcmalloc)
+#     list(APPEND package_deps tcmalloc)
+# endif()
+set(EVHTP_DISABLE_REGEX ON)
+
+add_library(evhtp ${LIBEVHTP_SOURCE_FILES})
+target_link_libraries(evhtp PUBLIC ${LIBEVHTP_EXTERNAL_LIBS})
+
+target_compile_definitions(evhtp PUBLIC "-DPROJECT_VERSION=${PROJECT_VERSION}")
```

Comment:
Actually it will work:
* https://gitlab.kitware.com/cmake/cmake/merge_requests/1873

But I think it's better to remove it anyway, little bit confusing.

Reply:
Already done :)

Reply:
I had a feeling it worked because I have done this in the past without issue. Unclear from docs if that is actually supported.